src/nvim/linematch.c

```diff
@@ -150,7 +150,9 @@ static int count_n_matched_chars(mmfile_t **sp, const size_t n, bool iwhite)
 mmfile_t fastforward_buf_to_lnum(mmfile_t s, linenr_T lnum)
 {
   for (int i = 0; i < lnum - 1; i++) {
-    s.ptr = strnchr(s.ptr, (size_t *)&s.size, '\n');
+    size_t n = (size_t)s.size;
+    s.ptr = strnchr(s.ptr, &n, '\n');
+    s.size = n;
```

Comment:
```suggestion
    s.size = (int)n;
```

Reply:
Yeah I see that it failed the checks because of that but I'm a bit confused why my machine isn't failing to compile. I tried make lint and compiling with gcc, clang, and zig cc and they all allow it for some reason but the checks here fail.

Reply:
Because `-Werror` is only used on CI

Reply:
Ah I see that makes sense, thanks!